// retell-sdk/resources/knowledge-base.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../resource.mjs";
import * as Core from "../core.mjs";
export class KnowledgeBase extends APIResource {
    /**
     * Create a new knowledge base
     */
    create(body, options) {
        return this._client.post('/create-knowledge-base', Core.multipartFormRequestOptions({ body, ...options }));
    }
    /**
     * Retrieve details of a specific knowledge base
     */
    retrieve(knowledgeBaseId, options) {
        return this._client.get(`/get-knowledge-base/${knowledgeBaseId}`, options);
    }
    /**
     * List all knowledge bases
     */
    list(options) {
        return this._client.get('/list-knowledge-bases', options);
    }
    /**
     * Delete an existing knowledge base
     */
    delete(knowledgeBaseId, options) {
        return this._client.delete(`/delete-knowledge-base/${knowledgeBaseId}`, {
            ...options,
            headers: { Accept: '*/*', ...options?.headers },
        });
    }
    /**
     * Add sources to a knowledge base
     */
    addSources(knowledgeBaseId, body, options) {
        return this._client.post(`/add-knowledge-base-sources/${knowledgeBaseId}`, Core.multipartFormRequestOptions({ body, ...options }));
    }
    /**
     * Delete an existing source from knowledge base
     */
    deleteSource(knowledgeBaseId, sourceId, options) {
        return this._client.delete(`/delete-knowledge-base-source/${knowledgeBaseId}/source/${sourceId}`, options);
    }
}
