// retell-sdk/resources/llm.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.Llm = void 0;
const resource_1 = require("../resource.js");
class Llm extends resource_1.APIResource {
    /**
     * Create a new Retell LLM Response Engine that can be attached to an agent. This
     * is used to generate response output for the agent.
     */
    create(body, options) {
        return this._client.post('/create-retell-llm', { body, ...options });
    }
    /**
     * Retrieve details of a specific Retell LLM Response Engine
     */
    retrieve(llmId, options) {
        return this._client.get(`/get-retell-llm/${llmId}`, options);
    }
    /**
     * Update an existing Retell LLM Response Engine
     */
    update(llmId, body, options) {
        return this._client.patch(`/update-retell-llm/${llmId}`, { body, ...options });
    }
    /**
     * List all Retell LLM Response Engines that can be attached to an agent.
     */
    list(options) {
        return this._client.get('/list-retell-llms', options);
    }
    /**
     * Delete an existing Retell LLM Response Engine
     */
    delete(llmId, options) {
        return this._client.delete(`/delete-retell-llm/${llmId}`, {
            ...options,
            headers: { Accept: '*/*', ...options?.headers },
        });
    }
}
exports.Llm = Llm;
