// retell-sdk/resources/phone-number.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../resource.mjs";
export class PhoneNumber extends APIResource {
    /**
     * Buy a new phone number & Bind agents
     */
    create(body, options) {
        return this._client.post('/create-phone-number', { body, ...options });
    }
    /**
     * Retrieve details of a specific phone number
     */
    retrieve(phoneNumber, options) {
        return this._client.get(`/get-phone-number/${phoneNumber}`, options);
    }
    /**
     * Update agent bound to a purchased phone number
     */
    update(phoneNumber, body, options) {
        return this._client.patch(`/update-phone-number/${phoneNumber}`, { body, ...options });
    }
    /**
     * List all phone numbers
     */
    list(options) {
        return this._client.get('/list-phone-numbers', options);
    }
    /**
     * Delete an existing phone number
     */
    delete(phoneNumber, options) {
        return this._client.delete(`/delete-phone-number/${phoneNumber}`, {
            ...options,
            headers: { Accept: '*/*', ...options?.headers },
        });
    }
    /**
     * Import a phone number from custom telephony & Bind agents
     */
    import(body, options) {
        return this._client.post('/import-phone-number', { body, ...options });
    }
}
