// retell-sdk/index.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var _a;
Object.defineProperty(exports, "__esModule", { value: true });
exports.UnprocessableEntityError = exports.PermissionDeniedError = exports.InternalServerError = exports.AuthenticationError = exports.BadRequestError = exports.RateLimitError = exports.ConflictError = exports.NotFoundError = exports.APIUserAbortError = exports.APIConnectionTimeoutError = exports.APIConnectionError = exports.APIError = exports.RetellError = exports.fileFromPath = exports.toFile = exports.Retell = void 0;
const Core = __importStar(require("./core.js"));
const Errors = __importStar(require("./error.js"));
const Uploads = __importStar(require("./uploads.js"));
const API = __importStar(require("./resources/index.js"));
const agent_1 = require("./resources/agent.js");
const batch_call_1 = require("./resources/batch-call.js");
const call_1 = require("./resources/call.js");
const concurrency_1 = require("./resources/concurrency.js");
const knowledge_base_1 = require("./resources/knowledge-base.js");
const llm_1 = require("./resources/llm.js");
const phone_number_1 = require("./resources/phone-number.js");
const voice_1 = require("./resources/voice.js");
const Webhooks = __importStar(require("retell-sdk/lib/webhook_auth"));
/**
 * API Client for interfacing with the Retell API.
 */
class Retell extends Core.APIClient {
    /**
     * API Client for interfacing with the Retell API.
     *
     * @param {string} opts.apiKey
     * @param {string} [opts.baseURL=process.env['RETELL_BASE_URL'] ?? https://api.retellai.com] - Override the default base URL for the API.
     * @param {number} [opts.timeout=1 minute] - The maximum amount of time (in milliseconds) the client will wait for a response before timing out.
     * @param {number} [opts.httpAgent] - An HTTP agent used to manage HTTP(s) connections.
     * @param {Core.Fetch} [opts.fetch] - Specify a custom `fetch` function implementation.
     * @param {number} [opts.maxRetries=2] - The maximum number of times the client will retry a request.
     * @param {Core.Headers} opts.defaultHeaders - Default headers to include with every request to the API.
     * @param {Core.DefaultQuery} opts.defaultQuery - Default query parameters to include with every request to the API.
     */
    constructor({ baseURL = Core.readEnv('RETELL_BASE_URL'), apiKey, ...opts }) {
        if (apiKey === undefined) {
            throw new Errors.RetellError("Missing required client option apiKey; you need to instantiate the Retell client with an apiKey option, like new Retell({ apiKey: 'YOUR_RETELL_API_KEY' }).");
        }
        const options = {
            apiKey,
            ...opts,
            baseURL: baseURL || `https://api.retellai.com`,
        };
        super({
            baseURL: options.baseURL,
            timeout: options.timeout ?? 60000 /* 1 minute */,
            httpAgent: options.httpAgent,
            maxRetries: options.maxRetries,
            fetch: options.fetch,
        });
        this.call = new API.Call(this);
        this.phoneNumber = new API.PhoneNumber(this);
        this.agent = new API.Agent(this);
        this.llm = new API.Llm(this);
        this.knowledgeBase = new API.KnowledgeBase(this);
        this.voice = new API.Voice(this);
        this.concurrency = new API.Concurrency(this);
        this.batchCall = new API.BatchCall(this);
        this._options = options;
        this.apiKey = apiKey;
    }
    defaultQuery() {
        return this._options.defaultQuery;
    }
    defaultHeaders(opts) {
        return {
            ...super.defaultHeaders(opts),
            ...this._options.defaultHeaders,
        };
    }
    authHeaders(opts) {
        return { Authorization: `Bearer ${this.apiKey}` };
    }
}
exports.Retell = Retell;
_a = Retell;
Retell.Retell = _a;
Retell.DEFAULT_TIMEOUT = 60000; // 1 minute
Retell.RetellError = Errors.RetellError;
Retell.APIError = Errors.APIError;
Retell.APIConnectionError = Errors.APIConnectionError;
Retell.APIConnectionTimeoutError = Errors.APIConnectionTimeoutError;
Retell.APIUserAbortError = Errors.APIUserAbortError;
Retell.NotFoundError = Errors.NotFoundError;
Retell.ConflictError = Errors.ConflictError;
Retell.RateLimitError = Errors.RateLimitError;
Retell.BadRequestError = Errors.BadRequestError;
Retell.AuthenticationError = Errors.AuthenticationError;
Retell.InternalServerError = Errors.InternalServerError;
Retell.PermissionDeniedError = Errors.PermissionDeniedError;
Retell.UnprocessableEntityError = Errors.UnprocessableEntityError;
Retell.toFile = Uploads.toFile;
Retell.fileFromPath = Uploads.fileFromPath;
Retell.verify = Webhooks.verify;
Retell.sign = Webhooks.sign;
Retell.Call = call_1.Call;
Retell.PhoneNumber = phone_number_1.PhoneNumber;
Retell.Agent = agent_1.Agent;
Retell.Llm = llm_1.Llm;
Retell.KnowledgeBase = knowledge_base_1.KnowledgeBase;
Retell.Voice = voice_1.Voice;
Retell.Concurrency = concurrency_1.Concurrency;
Retell.BatchCall = batch_call_1.BatchCall;
var uploads_1 = require("./uploads.js");
Object.defineProperty(exports, "toFile", { enumerable: true, get: function () { return uploads_1.toFile; } });
Object.defineProperty(exports, "fileFromPath", { enumerable: true, get: function () { return uploads_1.fileFromPath; } });
var error_1 = require("./error.js");
Object.defineProperty(exports, "RetellError", { enumerable: true, get: function () { return error_1.RetellError; } });
Object.defineProperty(exports, "APIError", { enumerable: true, get: function () { return error_1.APIError; } });
Object.defineProperty(exports, "APIConnectionError", { enumerable: true, get: function () { return error_1.APIConnectionError; } });
Object.defineProperty(exports, "APIConnectionTimeoutError", { enumerable: true, get: function () { return error_1.APIConnectionTimeoutError; } });
Object.defineProperty(exports, "APIUserAbortError", { enumerable: true, get: function () { return error_1.APIUserAbortError; } });
Object.defineProperty(exports, "NotFoundError", { enumerable: true, get: function () { return error_1.NotFoundError; } });
Object.defineProperty(exports, "ConflictError", { enumerable: true, get: function () { return error_1.ConflictError; } });
Object.defineProperty(exports, "RateLimitError", { enumerable: true, get: function () { return error_1.RateLimitError; } });
Object.defineProperty(exports, "BadRequestError", { enumerable: true, get: function () { return error_1.BadRequestError; } });
Object.defineProperty(exports, "AuthenticationError", { enumerable: true, get: function () { return error_1.AuthenticationError; } });
Object.defineProperty(exports, "InternalServerError", { enumerable: true, get: function () { return error_1.InternalServerError; } });
Object.defineProperty(exports, "PermissionDeniedError", { enumerable: true, get: function () { return error_1.PermissionDeniedError; } });
Object.defineProperty(exports, "UnprocessableEntityError", { enumerable: true, get: function () { return error_1.UnprocessableEntityError; } });
exports = module.exports = Retell;
exports.default = Retell;
