// retell-sdk/lib/webhook_auth.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.sign = exports.verify = exports.combined = exports.asymmetric = exports.symmetric = void 0;
const crypto_1 = require("crypto");
const FIVE_MINUTES = 5 * 60 * 1000;
function makeSecureWebhooks(getSigner, getVerifier) {
    return {
        sign(input, secretOrPrivateKey, timestamp = Date.now()) {
            const signer = getSigner(secretOrPrivateKey);
            return `v=${timestamp},d=${signer(input + timestamp)}`;
        },
        verify(input, secret, signature, opts = {}) {
            const match = /v=(\d+),d=(.*)/.exec(signature);
            if (!match) {
                return false;
            }
            const poststamp = Number(match[1]);
            const postDigest = match[2];
            const timestamp = opts?.timestamp ?? Date.now();
            const timeout = opts?.timeout ?? FIVE_MINUTES;
            const difference = Math.abs(timestamp - poststamp);
            if (difference > timeout) {
                return false;
            }
            const verifier = getVerifier(secret);
            return verifier(input + poststamp, postDigest);
        },
    };
}
exports.symmetric = makeSecureWebhooks((secret) => (input) => (0, crypto_1.createHmac)('sha256', secret).update(input).digest('hex'), (secret) => (input, digest) => (0, crypto_1.createHmac)('sha256', secret).update(input).digest('hex') === digest);
exports.asymmetric = makeSecureWebhooks((priv) => (input) => (0, crypto_1.createSign)('sha256').update(input).sign(priv, 'base64'), (pub) => (input, digest) => (0, crypto_1.createVerify)('sha256').update(input).verify(pub, digest, 'base64'));
exports.combined = {
    sign: (input, secretOrPrivateKey, timestamp) => secretOrPrivateKey.includes('PRIVATE KEY') ?
        exports.asymmetric.sign(input, secretOrPrivateKey, timestamp)
        : exports.symmetric.sign(input, secretOrPrivateKey, timestamp),
    verify: (input, secretOrPublicKey, signature, opts) => secretOrPublicKey.includes('PUBLIC KEY') ?
        exports.asymmetric.verify(input, secretOrPublicKey, signature, opts)
        : exports.symmetric.verify(input, secretOrPublicKey, signature, opts),
};
// export const sign = symmetric.sign;
// export const verify = symmetric.verify;
const verify = (body, apiKey, signature) => {
    return exports.symmetric.verify(body, apiKey, signature);
};
exports.verify = verify;
const sign = (body, apiKey) => {
    return exports.symmetric.sign(body, apiKey);
};
exports.sign = sign;
